### PR TITLE
Feature/52595 ps report log download

### PR DESCRIPTION
**Pull request by GuyHarwood**
tech support pages to browse and download ps report logs from azure storage

### CODE REVIEW

File: admin/controllers/tech-support.js [hunk 1]
```diff
@@ -507,6 +508,49 @@ const controller = {
     } catch (error) {
       return next(error)
     }
+  },
+
+  getPsReportLogs: async function getPsReportLogs (req, res, next) {
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:513]: `getPsReportLogs`: suggest we rename this to `getPsReportFoldersPage`, or perhaps, `getPsReportFolders` if you think the `Page` suffix is not useful.

File: admin/controllers/tech-support.js [hunk 2]
```diff
@@ -507,6 +508,49 @@ const controller = {
...
+      res.render('tech-support/ps-report-logs', {
+        breadcrumbs: req.breadcrumbs(),
+        logs: logs
+      })
+    } catch (error) {
+      return next(error)
+    }
+  },
+
+  getPsReportLogsFileList: async function getPsReportLogsFileList (req, res, next) {
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:527]: Suggest we rename `getPsReportLogsFileList` to `getPsReportFolderFileList` as I admit I was a bit confused on reading this as to the difference between this page and the previous one.

**Review by jon-shipley — APPROVED** [review]
LGTM 👍 

Suggested rename to the controller methods, should you agree.

File: admin/services/tech-support/ps-report-logs.service/data-access/ps-report-logs.data.service.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    const blobBuffer = await blobClient.downloadToBuffer()
+    return blobBuffer.toString()
+  }
+
+  public static async getContainerFileList (containerName: string): Promise<string[]> {
+    const client = BlobServiceClient.fromConnectionString(config.AZURE_STORAGE_CONNECTION_STRING)
+    const containerClient = await client.getContainerClient(containerName)
+    if (!containerClient.exists()) return []
+    const files = new Array<string>()
+    for await (const file of containerClient.listBlobsFlat({ includeMetadata: false, includeDeleted: false })) {
```
Comment (jon-shipley) [inline admin/services/tech-support/ps-report-logs.service/data-access/ps-report-logs.data.service.ts:36]: I am surprised we are not getting the metadata.  Does this mean you are not showing the file size to the user next to the download link?

Comment (GuyHarwood) [reply admin/services/tech-support/ps-report-logs.service/data-access/ps-report-logs.data.service.ts:36]: yes that's my plan.  i realise now i've not put this up as a draft 🤦🏼

File: admin/controllers/tech-support.js [hunk 4]
```diff
@@ -507,6 +508,50 @@ const controller = {
...
+    } catch (error) {
+      return next(error)
+    }
+  },
+
+  getPsReportLogFileContents: async function getPsReportLogFileContents (req, res, next) {
+    try {
+      const file = await psReportLogsDownloadService.downloadLogFile(req.params.folder, req.params.file)
+      res.set({
+        'Content-Disposition': `attachment; filename="${req.params.file}"`,
```
Comment (jon-shipley) [inline admin/controllers/tech-support.js:547]: Are there any security implications in trusting the Request parameter here?

Comment (GuyHarwood) [reply admin/controllers/tech-support.js:547]: good point.  these need sanitising.  will add some validation and narrowing checks

**Pull request merged**